src/components/Box/Box.service.tsx

```diff
@@ -0,0 +1,59 @@
+import type { CSSProperties } from "react";
+
+import type { BoxMarginPadding } from "./BoxTypes";
+import type { Theme } from "../utilities/theme";
+
+export const isSpaceValue = (value: any): value is number =>
+  typeof value === "number" && !Number.isNaN(value);
+
+export const getMarginPaddingStyles = (
```

Comment:
I tried a more clever solution with something like:

```ts
const spaceDirections = {
  ma: "margin",
  mb: "marginBottom",
  ml: "marginLeft",
  mr: "marginRight",
  // ...
};
```

…and then doing hash map lookups. It was a tiny bit bigger than this destructuring solution.